Update implicit casting example (#2370)
scala/docs.scala-lang: _overviews/scala3-book/first-look-at-types.md

For example:

```scala
val b: Byte = 127
val i: Int = b  // 127

val face: Char = '☺'
val number: Int = face  // 9786
```

You can only cast to a type if there is no loss of information. Otherwise, you need to be explicit about the cast:

```scala
val x: Long = 987654321

> val y: Float = x.toFloat  // 9.8765434E8 (note that `.toFloat` is required because it results in percision loss)

Reviewer: ```suggestion
val y: Float = x.toFloat  // 9.8765434E8 (note that `.toFloat` is required because the cast results in percision loss)
```

For example:

```scala
val b: Byte = 127
val i: Int = b  // 127

val face: Char = '☺'
val number: Int = face  // 9786
```

You can only cast to a type if there is no loss of information. Otherwise, you need to be explicit about the cast:

> This will not compile:

Reviewer: ```suggestion
```

```scala
val x: Long = 987654321

> val y: Float = x.toFloat  // 9.8765434E8 (note implicit conversion is depreciated since 2.13.1 when it would result in percision loss)

Reviewer: Thank you for the fix. There is typo in “deprec*i*ated” :)

However, I think this example should be moved below. It is better to first show the cast from `Char` to `Int`, which corresponds to a simple case of type widening.

We could maybe add another example (e.g. `Float` to `Double`?).

Then, there is this section below (from line 248 to 255) where we explain that casting to a narrower type does not work. I think we should move the `Long` to `Float` example here, and we should mention that it is also possible to cast from `Float` to `Long` by explicitly calling `toLong`.

Author: Great suggestions and feedback, thank you so much, and great catch on the typo.

I've updated, taking your feedback into consideration. I opted for a Byte -> Int example vs the suggested Float -> Double to avoid any confusion the increased precision might cause.
```scala
val f: Float = 3.14
val d: Double = f // 3.140000104904175
```

> Casting is unidirectional.

Reviewer: ```suggestion
You can only cast to a type if there is no loss of information. Otherwise, you need to be explicit about the cast:
```

This will not compile:

```scala
val x: Long = 987654321

> val y: Float = x.toFloat  // 9.8765434E8 (note implicit conversion is deprecated since 2.13.1, when it would result in percision loss)

Reviewer: ```suggestion
val y: Float = x.toFloat  // 9.8765434E8 (note that `.toFloat` is required because it results in percision loss)
```

> val z: Long = y  // Does not conform

Reviewer: ```suggestion
val z: Long = y  // Error
```
